mysite/polls/views.py

```diff
@@ -0,0 +1,43 @@
+from django.shortcuts import render
+import search
+import json
+from django.http import JsonResponse
+from django.http import HttpResponse
+import re
+# Create your views here.
+
+
+def make_link_for_html(movie_id):
+    link = '<a href = "http://localhost:8000/polls/info/?q=' \
+            + movie_id + '"> Link </a>'
+    return link
+
+
+def make_link(movie_id):
+    link = 'http://localhost:8000/polls/info/?q=' \
+            + movie_id  
+    return link
+
+
+def index(request):
+    ans = request.GET.get('q', 'Nothing')
+    movie_ids, movie_names, id_name_map = search.get_movie_names(ans)
+    print movie_ids
+    answer = ""
+    json_answer = {}
+    for movie_index in range(0, len(movie_ids)):
```

Comment:
Since you need to iterate over both `movie_names` and `movie_ids` togehter, you should look into `zip` function in python. The resulting code will be much cleaner.